server/render/index.js

```diff
@@ -145,26 +131,17 @@ export function serverRender( req, res ) {
 		links = [],
 		cacheKey = false;
 
-	if (
-		isSectionIsomorphic( context.store.getState() ) &&
-		! context.user &&
-		isCacheableQuery( context.query, context.cacheQueryKeys )
-	) {
-		cacheKey = getNormalizedPath( context.pathname, context.query );
-	}
-
 	if ( ! isDefaultLocale( context.lang ) ) {
 		const langFileName = getCurrentLocaleVariant( context.store.getState() ) || context.lang;
 		context.i18nLocaleScript = '//widgets.wp.com/languages/calypso/' + langFileName + '.js';
 	}
 
 	if (
 		config.isEnabled( 'server-side-rendering' ) &&
-		context.layout &&
-		! context.user &&
-		cacheKey &&
-		isDefaultLocale( context.lang )
+		isServerSideRenderCandidate( context ) &&
```

Comment:
Should this function just return the result of all of the conditions here?

Reply:
Depends, it's out choice how we group these conditions. If we decide to use only one function, then I wouldn't call it `isServerSideRenderCandidate`, but rather just `shouldServerSideRender`.

The current `isServerSideRenderCandidate` focuses on the app-checks directly related to the context (while doSSR would be more query related, and config is a global config). I guess the best solution (which I'll push) is to keep the current `isServerSideRenderCandidate` and add a new one `shouldServerSideRender` that includes all 3 conditions.